src/main/java/com/powsybl/openloadflow/ac/ReactiveLimitsOuterLoop.java

```diff
@@ -63,35 +70,83 @@ private void switchPvPq(LfBus bus, EquationSystem equationSystem, VariableSet va
         }
     }
 
+    private enum ReactiveLimitDirection {
+        MIN,
+        MAX
+    }
+
+    private static final class PvToPqBus {
+
+        private final LfBus bus;
+
+        private final double q;
+
+        private final double qLimit;
+
+        private final ReactiveLimitDirection limitDirection;
+
+        private PvToPqBus(LfBus bus, double q, double qLimit, ReactiveLimitDirection limitDirection) {
+            this.bus = bus;
+            this.q = q;
+            this.qLimit = qLimit;
+            this.limitDirection = limitDirection;
+        }
+    }
+
+    private void switchPvPq(List<PvToPqBus> pvToPqBuses, EquationSystem equationSystem, VariableSet variableSet, int remainingPvBusCount) {
+        int modifiedRemainingPvBusCount = remainingPvBusCount;
+        if (modifiedRemainingPvBusCount == 0) {
```

Comment:
You need to add a test before the if statement ````if (pvToPqBuses.size() > 1) {...```` in order not to remove a PV bus when juste one PV bus is switching.

Reply:
To void infinite loop I implemented another strategy: just consider outer loop stable if just one PV bus should have been switched PQ but not.